### ISSUE TITLE
update jquery to 3.x in default theme

### ISSUE THREAD

**Issue by aciccarello** [TypeStrong/typedoc#994]
- [x] I have checked [issues with enhancement label](https://github.com/TypeStrong/typedoc/labels/enhancement) and found no duplicates

## Problem
The jquery dependency we use has some known security issues. These weren't previously recognized because the dependency was not declared. The fix is non-trivial because of some strict compilation errors.

This would be a PR on TypeStrong/typedoc-default-themes

**Comment by Gerrit0** [comment]
Reopening as the fix isn't present in this repository yet.

**Comment by mcparlandjcgi** [comment]
Is there a timescale for this being released to the repository?

**Comment by AndrewCraswell** [comment]
Commenting here because #1009 and others identifying the `marked` security vulnerability have been closed in favor of this issue, but it does not specifically mention `marked` here so I'm afraid this task has gone missing... It's been nearly a month since `marked` started identifying it needs to be patched to version `0.6.2` to fix the vulnerability. Is there a timeline on this? I'm going ahead and removing TypeDc from my projects for now since it seems like it could be awhile.

Unfortunately, this is the second time in the last 3 months I've had to remove TypeDoc because security vulnerabilities are taking too long to get patched. It seems in the context of this particular issue (#994) the update of jQuery was nontrivial, which is understandable. But maybe these security fixes shouldn't be group together and would be better off being treated as separate issues?
